mitmproxy/optmanager.py

```diff
@@ -365,6 +357,30 @@ def parse_setval(self, o: _Option, optstr: typing.Optional[str], currentvalue: t
                     return [optstr]
         raise NotImplementedError("Unsupported option type: %s", o.typespec)
 
+    @staticmethod
+    def _parse_numeric(optstr: typing.Optional[str], name: str,
+                       typespec: type) -> typing.Union[int, float, None]:
+        """Helper method to parse a numeric option value from a string"""
+        is_optional = typespec in (typing.Optional[int], typing.Optional[float])
+        if optstr:
+            try:
+                if typespec in (int, typing.Optional[int]):
+                    return int(optstr)
+                elif typespec in (float, typing.Optional[float]):
+                    return float(optstr)
+                else:
+                    return None
+            except ValueError:
+                if not is_optional:
+                    typename = str(typespec).split("'")[1]
+                    raise exceptions.OptionsError(f"Not an {typename}: {optstr}")
+                else:
+                    return None
```

Comment:
Do we intentionally discard invalid values for optionals here? Just from the source it looks like they should raise an error as well?